frontend/src/components/fields/amount/Amount.tsx

```diff
@@ -0,0 +1,34 @@
+import React, {useState} from 'react';
+import {AmountProps} from "./props";
+import {TextField} from "@mui/material";
+import {ValidationErrorMessages} from "../../../common/constansts/validationErrorMessages";
+
+const Amount: React.FC<AmountProps> = ({updateAmountFields}) => {
+    const [amount, setAmount] = useState<number>(0);
+    const [isAmountValid, setIsAmountValid] = useState<boolean>(false);
+
+    const handleAmountChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const inputValue = parseFloat(e.target.value);
+        setAmount(inputValue);
+        const isValid = validateAmount(inputValue);
+        setIsAmountValid(isValid);
+        updateAmountFields(amount, isValid);
+    };
```

Comment:
make the whole process a bit different. Instead of allowing to type incorrect number and check correctness afterwards, do the opposite. 
1. Check the number, if it's incorrect, don't set number
2. If ok, set number and pass to parent component

In that case you don't need  const [isAmountValid, setIsAmountValid] = useState<boolean>(false);

Reply:
if i remove isAmountValid, how do i validate input?

Reply:
Your state will be always in valid state, because you set only correct numbers

Reply:
it means, that if user reached upper limit i just disable button without message, am i right?

Reply:
You show helper message all the time

Reply:
ok, in that case i should first send isValid to `PaymentEntry` to able/disable button and than i send amount to `PaymentEntry` based on isValid variable, am i right?

Reply:
Keep it able all the time

Reply:
what's trouble to do like i wrote? Is it bad approach (i think not) or we don't care about what number user write? i don't understand

Reply:
I think, if we can simplify something, it's better to do it
```typescript
const [amount, setAmount] = useState<number>(0);

    const handleAmountChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        const inputValue = parseFloat(e.target.value);
        if (!validateAmount(inputValue)) return;
        setAmount(inputValue);
        updateAmountFields(amount);
    };

    const validateAmount = (amount: number): boolean => amount > 0 && amount < 99999999;

    return (
        <TextField
            helperText={ValidationErrorMessages.AMOUNT_NOT_VALID}
            placeholder="Amount"
            variant="outlined"
            type="number"
            onChange={handleAmountChange}
            fullWidth
            margin="normal"
        />
    );
``` 
If you wish, you can leave as it is